Sources/BSON/Codable/Decoding/BSONDecoderSettings.swift

```diff
@@ -169,6 +193,12 @@ public struct BSONDecoderSettings {
     
     /// If `true`, allows decoding ObjectIds from Strings if they're formatted as a 24-character hexString
     public var decodeObjectIdFromString: Bool = false
+
+    /// If `true`, allows decoding Date from a Double (TimeInterval)
+    public var decodeDateFromTimestamp: Bool {
+        get { timestampToDateDecodingStrategy != .never }
+        set { timestampToDateDecodingStrategy = newValue ? .relativeToUnixEpoch : .never }
```

Comment:
As per the setter implementation, if anyone has opted for `timestampToDateDecodingStrategy` as .relativeToReferenceDate, and set this `decodeDateFromTimestamp` as true without looking into the implementation, the strategy will get reset to `.relativeToUnixEpoch`. Might be confusing to understand the impact.

set { 
    if !newValue { timestampToDateDecodingStrategy = .never }
    else if timestampToDateDecodingStrategy == .never { timestampToDateDecodingStrategy = .relativeToUnixEpoch }
    // else case will keep the date decoding strategy to whatever value it had before